keras_cv/visualization/utils.py

```diff
@@ -0,0 +1,56 @@
+from typing import Dict
+from typing import List
+from typing import Tuple
+from typing import Union
+
+import tensorflow as tf
+
+E = Union[str, Dict[str, Union[str, int]]]
+
+
+def get_nested_layer(model: tf.keras.Model, name: str) -> tf.keras.layers.Layer:
+    """Get nested layer in model.
+
+    Example:
+    ```py
+    model = tf.keras.Sequential([
+        tf.keras.applications.ResNet101V2(include_top=False, pooling='avg'),
+        tf.keras.layers.Dense(10, activation='softmax', name='predictions')
+    ])
+
+    pooling_layer = get_nested_layer(model, 'resnet101v2.avg_pool')
+    ```
+    """
+    for n in name.split("."):
+        model = model.get_layer(n)
+
+    return model
+
+
+def collect_endpoints(model: tf.keras.Model, endpoints: List[E]) -> List[tf.Tensor]:
+    """Collect (intermediate) endpoints in a model."""
+    endpoints_ = []
+
+    for ep in endpoints:
+        if isinstance(ep, str):
+            ep = {"name": ep}
+
+        layer = ep["name"]
+        link = ep.get("link", "output")
+        bind = ep.get("bind", 0)
+
+        endpoints_.append(
+            get_nested_layer(model, layer).get_input_at(bind)
+            if link == "input"
+            else get_nested_layer(model, layer).get_output_at(bind)
+        )
+
+    return endpoints_
+
+
+def normalize(x: tf.Tensor, axis: Tuple[int, int] = (-3, -2)) -> tf.Tensor:
```

Comment:
Let's use `utils.transform_value_range()` instead

Reply:
I don't think it's possible for two reasons:
1. `transform_value_range` requires the original range, which is unknown (the signal from the last conv. layer and from logits might be arbitrarily large)
2. I want to normalize the activation map for each class independently, thus using `axis=(-3, -2)=(height, width)`; but `transform_value_range` was clearly developed to normalize with respect to a single pair (min, max) of numbers.